app/assets/javascripts/discourse/controllers/topic.js.es6

```diff
@@ -679,6 +679,24 @@ export default Controller.extend(bufferedProperty("model"), {
       }
     },
 
+    toggleFeaturedOnProfile() {
+      if (!this.currentUser) return;
+      const toggle = () => {
```

Comment:
I don't think this needs to be a closure since it's not wrapping any variables in this scope. We'd use less memory by making it a method on the controller like `_performToggleFeatured`